corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -996,6 +996,59 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
     VideoEntry.prototype = Object.create(FileEntry.prototype);
     VideoEntry.prototype.constructor = FileEntry;
 
+    /**
+     * Represents a signature, which requires the user to draw a signature.
+     */
+    function SignatureEntry(question, options) {
+        var self = this;
+        FileEntry.call(this, question, options);
+        self.templateType = 'signature';
+        self.accept = 'image/*,.pdf';
+
+        self.afterRender = function () {
+            self.$input = $('#' + self.entryId);
+            self.$canvas = $('#' + self.entryId + '-canvas');
+            self.$wrapper = $('#' + self.entryId + '-wrapper');
+
+            self.signaturePad = new SignaturePad(self.$canvas[0]);
+            self.signaturePad.addEventListener('endStroke', () => { self.answerCanvasData(); });
+
+            new ResizeObserver(() => {
+                self.resizeCanvas();
+            }).observe(self.$wrapper[0]);
+
+            self.resizeCanvas();
+        };
+
+        self.answerCanvasData = function () {
+            self.$canvas[0].toBlob(blob => {
+                var filename = blob.size + '.png', // simple filename change for validation
+                    signatureFile = new File([blob], filename, {type: "image/png"}),
+                    list = new DataTransfer();
+                list.items.add(signatureFile);
+                self.$input[0].files = list.files;
+                self.rawAnswer(constants.FILE_PREFIX + filename);
+            });
+        };
+
+        self.onClear = function () {
+            // TODO: Connect this to the imminent "clear" action on FileEntry
+            self.signaturePad.clear();
+        };
+
+        self.resizeCanvas = function () {
+            var aspectRatio = 4,
+                width = self.$wrapper.width() - 2; // otherwise misaligned by 2px
+            self.$canvas[0].width = width;
+            self.$canvas[0].height = width / aspectRatio;
+        };
+
+        self.helpText = function () {
+            return gettext("Draw a signature");
```

Comment:
maybe "Draw your signature"?

Reply:
I like "sign here" personally but I think it's more directive than informational. Maybe just "Draw signature"? That would go with "Upload image" etc.